{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import statsmodels.api as sm \n",
    "import statsmodels.formula.api as smf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Visualize genetic relatedness between the strains by performing principal component analysis and plotting the first two components."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90p 64-bit (16 Jun 2020)              www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --mind\n",
      "  --pca\n",
      "  --vcf BYxRM_segs_saccer3.bam.simplified.vcf\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: plink-temporary.bed + plink-temporary.bim + plink-temporary.fam written.\n",
      "16624 variants loaded from .bim file.\n",
      "1056 people (0 males, 0 females, 1056 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to plink.nosex .\n",
      "105 people removed due to missing genotype data (--mind).\n",
      "IDs written to plink.irem .\n",
      "Using up to 4 threads (change this with --threads).\n",
      "Before main variant filters, 951 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate in remaining samples is 0.970565.\n",
      "16624 variants and 951 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "Excluding 1515 variants on non-autosomes from relationship matrix calc.\n",
      "\r",
      "60 markers complete.\r",
      "120 markers complete.\r",
      "180 markers complete.\r",
      "240 markers complete.\r",
      "300 markers complete.\r",
      "360 markers complete.\r",
      "420 markers complete.\r",
      "480 markers complete.\r",
      "540 markers complete.\r",
      "600 markers complete.\r",
      "660 markers complete.\r",
      "720 markers complete.\r",
      "780 markers complete.\r",
      "840 markers complete.\r",
      "900 markers complete.\r",
      "960 markers complete.\r",
      "1020 markers complete.\r",
      "1080 markers complete.\r",
      "1140 markers complete.\r",
      "1200 markers complete.\r",
      "1260 markers complete.\r",
      "1320 markers complete.\r",
      "1380 markers complete.\r",
      "1440 markers complete.\r",
      "1500 markers complete.\r",
      "1560 markers complete.\r",
      "1620 markers complete.\r",
      "1680 markers complete.\r",
      "1740 markers complete.\r",
      "1800 markers complete.\r",
      "1860 markers complete.\r",
      "1920 markers complete.\r",
      "1980 markers complete.\r",
      "2040 markers complete.\r",
      "2100 markers complete.\r",
      "2160 markers complete.\r",
      "2220 markers complete.\r",
      "2280 markers complete.\r",
      "2340 markers complete.\r",
      "2400 markers complete.\r",
      "2460 markers complete.\r",
      "2520 markers complete.\r",
      "2580 markers complete.\r",
      "2640 markers complete.\r",
      "2700 markers complete.\r",
      "2760 markers complete.\r",
      "2820 markers complete.\r",
      "2880 markers complete.\r",
      "2940 markers complete.\r",
      "3000 markers complete.\r",
      "3060 markers complete.\r",
      "3120 markers complete.\r",
      "3180 markers complete.\r",
      "3240 markers complete.\r",
      "3300 markers complete.\r",
      "3360 markers complete.\r",
      "3420 markers complete.\r",
      "3480 markers complete.\r",
      "3540 markers complete.\r",
      "3600 markers complete.\r",
      "3660 markers complete.\r",
      "3720 markers complete.\r",
      "3780 markers complete.\r",
      "3840 markers complete.\r",
      "3900 markers complete.\r",
      "3960 markers complete.\r",
      "4020 markers complete.\r",
      "4080 markers complete.\r",
      "4140 markers complete.\r",
      "4200 markers complete.\r",
      "4260 markers complete.\r",
      "4320 markers complete.\r",
      "4380 markers complete.\r",
      "4440 markers complete.\r",
      "4500 markers complete.\r",
      "4560 markers complete.\r",
      "4620 markers complete.\r",
      "4680 markers complete.\r",
      "4740 markers complete.\r",
      "4800 markers complete.\r",
      "4860 markers complete.\r",
      "4920 markers complete.\r",
      "4980 markers complete.\r",
      "5040 markers complete.\r",
      "5100 markers complete.\r",
      "5160 markers complete.\r",
      "5220 markers complete.\r",
      "5280 markers complete.\r",
      "5340 markers complete.\r",
      "5400 markers complete.\r",
      "5460 markers complete.\r",
      "5520 markers complete.\r",
      "5580 markers complete.\r",
      "5640 markers complete.\r",
      "5700 markers complete.\r",
      "5760 markers complete.\r",
      "5820 markers complete.\r",
      "5880 markers complete.\r",
      "5940 markers complete.\r",
      "6000 markers complete.\r",
      "6060 markers complete.\r",
      "6120 markers complete.\r",
      "6180 markers complete.\r",
      "6240 markers complete.\r",
      "6300 markers complete.\r",
      "6360 markers complete.\r",
      "6420 markers complete.\r",
      "6480 markers complete.\r",
      "6540 markers complete.\r",
      "6600 markers complete.\r",
      "6660 markers complete.\r",
      "6720 markers complete.\r",
      "6780 markers complete.\r",
      "6840 markers complete.\r",
      "6900 markers complete.\r",
      "6960 markers complete.\r",
      "7020 markers complete.\r",
      "7080 markers complete.\r",
      "7140 markers complete.\r",
      "7200 markers complete.\r",
      "7260 markers complete.\r",
      "7320 markers complete.\r",
      "7380 markers complete.\r",
      "7440 markers complete.\r",
      "7500 markers complete.\r",
      "7560 markers complete.\r",
      "7620 markers complete.\r",
      "7680 markers complete.\r",
      "7740 markers complete.\r",
      "7800 markers complete.\r",
      "7860 markers complete.\r",
      "7920 markers complete.\r",
      "7980 markers complete.\r",
      "8040 markers complete.\r",
      "8100 markers complete.\r",
      "8160 markers complete.\r",
      "8220 markers complete.\r",
      "8280 markers complete.\r",
      "8340 markers complete.\r",
      "8400 markers complete.\r",
      "8460 markers complete.\r",
      "8520 markers complete.\r",
      "8580 markers complete.\r",
      "8640 markers complete.\r",
      "8700 markers complete.\r",
      "8760 markers complete.\r",
      "8820 markers complete.\r",
      "8880 markers complete.\r",
      "8940 markers complete.\r",
      "9000 markers complete.\r",
      "9060 markers complete.\r",
      "9120 markers complete.\r",
      "9180 markers complete.\r",
      "9240 markers complete.\r",
      "9300 markers complete.\r",
      "9360 markers complete.\r",
      "9420 markers complete.\r",
      "9480 markers complete.\r",
      "9540 markers complete.\r",
      "9600 markers complete.\r",
      "9660 markers complete.\r",
      "9720 markers complete.\r",
      "9780 markers complete.\r",
      "9840 markers complete.\r",
      "9900 markers complete.\r",
      "9960 markers complete.\r",
      "10020 markers complete.\r",
      "10080 markers complete.\r",
      "10140 markers complete.\r",
      "10200 markers complete.\r",
      "10260 markers complete.\r",
      "10320 markers complete.\r",
      "10380 markers complete.\r",
      "10440 markers complete.\r",
      "10500 markers complete.\r",
      "10560 markers complete.\r",
      "10620 markers complete.\r",
      "10680 markers complete.\r",
      "10740 markers complete.\r",
      "10800 markers complete.\r",
      "10860 markers complete.\r",
      "10920 markers complete.\r",
      "10980 markers complete.\r",
      "11040 markers complete.\r",
      "11100 markers complete.\r",
      "11160 markers complete.\r",
      "11220 markers complete.\r",
      "11280 markers complete.\r",
      "11340 markers complete.\r",
      "11400 markers complete.\r",
      "11460 markers complete.\r",
      "11520 markers complete.\r",
      "11580 markers complete.\r",
      "11640 markers complete.\r",
      "11700 markers complete.\r",
      "11760 markers complete.\r",
      "11820 markers complete.\r",
      "11880 markers complete.\r",
      "11940 markers complete.\r",
      "12000 markers complete.\r",
      "12060 markers complete.\r",
      "12120 markers complete.\r",
      "12180 markers complete.\r",
      "12240 markers complete.\r",
      "12300 markers complete.\r",
      "12360 markers complete.\r",
      "12420 markers complete.\r",
      "12480 markers complete.\r",
      "12540 markers complete.\r",
      "12600 markers complete.\r",
      "12660 markers complete.\r",
      "12720 markers complete.\r",
      "12780 markers complete.\r",
      "12840 markers complete.\r",
      "12900 markers complete.\r",
      "12960 markers complete.\r",
      "13020 markers complete.\r",
      "13080 markers complete.\r",
      "13140 markers complete.\r",
      "13200 markers complete.\r",
      "13260 markers complete.\r",
      "13320 markers complete.\r",
      "13380 markers complete.\r",
      "13440 markers complete.\r",
      "13500 markers complete.\r",
      "13560 markers complete.\r",
      "13620 markers complete.\r",
      "13680 markers complete.\r",
      "13740 markers complete.\r",
      "13800 markers complete.\r",
      "13860 markers complete.\r",
      "13920 markers complete.\r",
      "13980 markers complete.\r",
      "14040 markers complete.\r",
      "14100 markers complete.\r",
      "14160 markers complete.\r",
      "14220 markers complete.\r",
      "14280 markers complete.\r",
      "14340 markers complete.\r",
      "14400 markers complete.\r",
      "14460 markers complete.\r",
      "14520 markers complete.\r",
      "14580 markers complete.\r",
      "14640 markers complete.\r",
      "14700 markers complete.\r",
      "14760 markers complete.\r",
      "14820 markers complete.\r",
      "14880 markers complete.\r",
      "14940 markers complete.\r",
      "15000 markers complete.\r",
      "15060 markers complete.\r",
      "15109 markers complete.\r",
      "Relationship matrix calculation complete.\n",
      "[extracting eigenvalues and eigenvectors]\r",
      "--pca: Results saved to plink.eigenval and plink.eigenvec .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "plink \\\n",
    "  --vcf BYxRM_segs_saccer3.bam.simplified.vcf \\\n",
    "  --pca \\\n",
    "  --mind \\\n",
    "  --allow-extra-chr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Visualize the allele frequency spectrum by plotting a histogram of allele frequencies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Using plink, perform quantitative association testing for all phenotypes. Use the top 10 principal components (eigenvectors) as covariates in your analysis, to adjust for non-independence due to relatedness.\n",
    "Be sure to use the --allow-no-sex option"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. For each, produce a QQ plot and Manhattan plot. For the Manhattan plot, highlight SNPs with p-values less than 10-5 in a different color.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Choose one of the traits for which you performed GWAS. For the top associated SNP, visualize the effect size by creating a boxplot of the phenotype stratified by genotype.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
